integration/tests/next-middleware.test.ts

```diff
@@ -0,0 +1,110 @@
+import { expect, test } from '@playwright/test';
+
+import type { Application } from '../models/application';
+import { appConfigs } from '../presets';
+import { createTestUtils } from '../testUtils';
+
+test.describe('next middleware @nextjs', () => {
+  test.describe.configure({ mode: 'parallel' });
+  let app: Application;
+
+  test.beforeAll(async () => {
+    app = await appConfigs.next.appRouter
+      .clone()
+      .addFile(
+        'src/middleware.ts',
+        () => `import { authMiddleware } from '@clerk/nextjs/server';
+import { NextResponse } from "next/server";
+
+export default authMiddleware({
+  publicRoutes: ['/', '/hash/sign-in', '/hash/sign-up'],
+  afterAuth: async (auth, req) => {
+    const response = NextResponse.next();
+    response.cookies.set({
+      name: "first",
+      value: "123456789",
+      path: "/",
+      sameSite: "none",
+      secure: true,
+    });
+    response.cookies.set("second", "987654321", {
+      sameSite: "none",
+      secure: true,
+    });
+    response.cookies.set("third", "foobar", {
+      sameSite: "none",
+      secure: true,
+    });
+    return response;
+  },
+});
+
+export const config = {
+  matcher: ['/((?!.*\\..*|_next).*)', '/', '/(api|trpc)(.*)'],
+};`,
+      )
+      .addFile(
+        'src/app/provider.tsx',
+        () => `'use client'
+import { ClerkProvider } from "@clerk/nextjs"
+
+export function Provider({ children }: { children: any }) {
+  return (
+    <ClerkProvider>
+      {children}
+    </ClerkProvider>
+  )
+}`,
+      )
+      .addFile(
+        'src/app/layout.tsx',
```

Comment:
The test **did not** run without these additions. The `next-build.test.ts` file has the same additions.

Why is the base template not just including these?